tests/python/unittest/test_gluon.py

```diff
@@ -3119,6 +3119,19 @@ def forward(self, x):
         shape = (np.random.randint(1, 10), np.random.randint(1, 10), 1)
         block(mx.nd.ones(shape))
 
+def test_shared_parameters_with_non_default_initializer():
+    class MyBlock(gluon.HybridBlock):
+        def __init__(self, **kwargs):
+            super(MyBlock, self).__init__(**kwargs)
+
+            with self.name_scope():
+                self.param = self.params.get("param", shape=(1, ), init=mx.init.Constant(-10.0))
+
+    bl = MyBlock()
+    bl2= MyBlock(params=bl.collect_params())
+    assert bl.param is bl2.param
```

Comment:
Thanks for your contribution!
Could you add the unittest 'b1.param is not bl3.param’ and the test for mx.init.Initializer?

Reply:
What do you mean? Actually `bl2 = MyBlock(params=bl.collect_params())` performs internally a check that the two initializes are equal, which will fail without this PR.